projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -101,6 +107,32 @@ export class IgxOverlayService implements OnDestroy {
         this._document = <Document>this.document;
     }
 
+    /**
+     * Generates Id. Provide this Id when call `show(id, settings?)` method
+     * @param component ElementRef or Component Type to show in overlay
+     * @param settings Display settings for the overlay, such as positioning and scroll/close behavior.
+     * @returns Id of the created overlay. Valid until `onClosed` is emitted.
+     */
+    register(component: ElementRef | Type<{}>, settings?: OverlaySettings): string {
+        let info: OverlayInfo;
+        const id = (this._componentId++).toString();
```

Comment:
Move the id generation after the return statement. So the componentId doesn't get increased if info is not correct

Reply:
done